ddp-workspace/projects/ddp-circadia/src/app/components/pages/dashboard/dashboard.component.ts

```diff
@@ -23,11 +27,29 @@ export class DashboardComponent implements OnInit {
   constructor(
     private router: Router,
     private userActivityService: UserActivityServiceAgent,
+    private announcementsService: AnnouncementsServiceAgent,
     @Inject('ddp.config') private config: ConfigurationService,
+    private sleepLogService: SleepLogService,
   ) {}
 
   ngOnInit(): void {
-    this.fetchActivities();
+    this.fetchData()
+      .pipe(
+        map(({ activities, announcements }) => ({
+          activities,
+          sleepLogCodes:
+            this.sleepLogService.extractSleepLogAnnouncements(announcements),
+        })),
+        concatMap(({ activities, sleepLogCodes }) =>
+          this.sleepLogService.runSleepLogActions(
+            sleepLogCodes,
+            !!activities.find(
+              activity => activity.activityCode === ActivityCode.MiniSleepLog,
+            ),
+          ),
+        ),
+      )
+      .subscribe();
```

Comment:
Above you can see this fragments of code that terminate subs:
```
...
this.subs.addNew(sub);
...
ngOnDestroy(): void {
    this.subs.removeAll();
}
```
In this case I 100% sure that this Observables emit values only one time, because they call services methods.